### DISCUSSION TITLE
Add mnamer for automatic media organization

### DISCUSSION THREAD

**Discussion by malfroid** [haugene/docker-transmission-openvpn#2013] [Ideas]
Feature request: add mnamer https://github.com/jkwill87/mnamer for automatic media organization after downloads complete.

**Comment by pkishino** [comment]
Uhm, had a brief look.. what does it offer that radarr/sonarr doesn’t already do and more??
Also, if you do want to use this, then just set it up separately. I don’t see any point in integration such a niche utility

**Comment by haugene** [comment]
Looks like a cool project and I guess this is more about avoiding radarr/sonarr setup than competing with it on features. If a simple tool can do the job, I see the appeal :smile: :+1: 
I do however agree with @pkishino that it seems a bit niche for us to include as an officially supported feature right now. In general we are trying to streamline the image a bit and do one thing, and do it well. We have gone a bit outside that path with including a proxy into the image - but that was by very popular demand and makes a bit of sense as well to keep the traffic proxy close to the VPN networking logic.

You could make this work either by just setting it up separately, or use the [custom scripts](https://haugene.github.io/docker-transmission-openvpn/config-options/#custom_scripts) feature to run a simple script that first installs pip, and then uses pip to install mnamer into the container on startup. From there you should be able to get this going using the on-complete script support in Transmission.

I think that would be the right way to go for now. If you proceed with this idea it would be great if you could share a write up of what you did to make it work and how you include it in your workflow. That could help others looking to do the same. And just ask if you have any problems getting it running.

**Comment by malfroid** [comment]
Thanks for the very constructive hints @haugene !

I was able to get this working with the following setup:

When launching the docker, make sure to bind a few more volumes and set some environment variables:

```yaml
transmission-openvpn:
container_name: transmission-openvpn
image: haugene/transmission-openvpn
volumes:
    - "/path/to/data:/data"
    - "/path/to/scripts:/scripts"
    - "/path/to/processed/tvshows:/tvshows"
    - "/path/to/processed/movies:/movies"
environment:
    - TRANSMISSION_SCRIPT_TORRENT_DONE_ENABLED=true
    - TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME=/scripts/mnamer.sh
```

Then in `/path/to/scripts`, place the following 4 files:

1. `openvpn-pre-start.sh`

```bash
#!/bin/bash

apt-get update
apt-get install python3-distutils -y
python3 /scripts/get-pip.py
pip3 install mnamer
```

2. `mnamer.sh`

```bash
#!/bin/bash

mnamer /data/completed/* --batch --config-path=/scripts/.mnamer-v2.json
```

3. `get-pip.py` - download script from [pip.pypa.io](https://pip.pypa.io/en/stable/installation/#get-pip-py)

4. `.mnamer-v2.json` - your mnamer configuration file. example:

```json
{
    "api_key_omdb": null,
    "api_key_tmdb": null,
    "api_key_tvdb": null,
    "api_key_tvmaze": null,
    "batch": false,
    "episode_api": "tvmaze",
    "episode_directory": "/tvshows/{series}/Season {season:02}",
    "episode_format": "{series} - S{season:02}E{episode:02} - {title}.{extension}",
    "hits": 5,
    "ignore": [
        ".*sample.*",
        "^RARBG.*"
    ],
    "language": null,
    "lower": false,
    "mask": [
        ".avi",
        ".m4v",
        ".mp4",
        ".mkv",
        ".ts",
        ".wmv",
        ".srt",
        ".idx",
        ".sub"
    ],
    "movie_api": "tmdb",
    "movie_directory": "/movies/{name} ({year})",
    "movie_format": "{name} ({year}).{extension}",
    "no_guess": false,
    "no_overwrite": true,
    "no_style": false,
    "recurse": true,
    "replace_after": {
        "&": "and",
        ";": ",",
        "@": "at"
    },
    "replace_before": {},
    "scene": false,
    "verbose": false
}
```
